rdt-open-reader/src/main/java/io/ona/rdt/util/Utils.java

```diff
@@ -102,9 +100,11 @@ public static int convertDpToPixels(Context context, float dp) {
         return (int) TypedValue.applyDimension(TypedValue.COMPLEX_UNIT_DIP, dp, context.getResources().getDisplayMetrics());
     }
 
-    public static void updateLocale(android.content.Context context, AllSharedPreferences allSharedPreferences) {
+    public static void updateLocale(Context context) {
 
-        Locale locale = new Locale(allSharedPreferences.fetchLanguagePreference());
+        String lang = RDTApplication.getInstance().getContext().allSharedPreferences().getPreference("locale");
+
+        Locale locale = new Locale(lang.length() > 0 ? lang : BuildConfig.LOCALE);
```

Comment:
`lang` can potentially be null here, use `StringUtils.isBlank(lang)` instead.

Reply:
fixed.

Reply:
Sorry, the comment here was to change this line to `Locale locale = new Locale(StringUtils.isNotBlank(lang) ? lang : BuildConfig.LOCALE);`.

Reply:
fixed.